specs/Cryptography/Ed25519.md

```diff
@@ -0,0 +1,69 @@
+# Ed25519
+
+Reference: https://tools.ietf.org/html/rfc8032#section-5.1
+
+## Public key
+```rust
+pub struct ED25519PublicKey(pub [u8; ed25519_dalek::PUBLIC_KEY_LENGTH]);
+```
+
+`ED25519PublicKey` is 32 bytes in size and represents a curve point in `CompressedEdwardsY` format:
+```rust
+/// In "Edwards y" / "Ed25519" format, the curve point \\((x,y)\\) is
+/// determined by the \\(y\\)-coordinate and the sign of \\(x\\).
+///
+/// The first 255 bits of a `CompressedEdwardsY` represent the
+/// \\(y\\)-coordinate.  The high bit of the 32nd byte gives the sign of \\(x\\).
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct CompressedEdwardsY(pub [u8; 32]);
```

Comment:
Still too much Rust. There is no such thing as "CompressedEdwardsY format" in the RFC.